Add document to setup SwiftLint for iOS swift apps

anand00 opened the pull request:
- Setup ``SwiftLint`` with basic iOS app in Swift.
- Considered Cocoapods for ``SwiftLint`` installations
- Used rules based on Raywenderlich Swift Style Guide 
- This configuration is tested on Swift 5.6.

anand00 committed:
Create swift-swiftLint.md

Add document to setup SwiftLint for Swift iOS applications

anand00 committed:
Delete swift-swiftLint.md

anand00 committed:
Add document to setup SwiftLint for iOS Swift apps

Abijeet commented on swift-swiftLint.md:
```diff
@@ -0,0 +1,131 @@
+# Introduction
+
+[SwiftLint](https://github.com/realm/SwiftLint/blob/master/README.md) is a tool by Realm to enforce Swift style and conventions. SwiftLint hooks into Apple’s own SourceKit framework to parse your Swift code, which means it supports the full range of Swift syntax and remains up to date as Swift continues to evolve.
+<br/>It is based on guidelines from the Swift style guide.
```
I would skip the `<br/>` tags and instead add just empty lines.

anand00 replied:
updated

Abijeet commented on swift-swiftLint.md:
````diff
@@ -0,0 +1,131 @@
+```
+
+# Usage
+If you've installed SwiftLint via CocoaPods, then 1. **Select Target** -> 2. **Build Phase** -> 3. **Click on +** -> 4. **Add New Run Script Phase** and <br /> copy & paste below script in Run Script
````
Can we put the items 1, 2, 3, in new lines?

anand00 replied:
updated

Abijeet commented on swift-swiftLint.md:
````diff
@@ -0,0 +1,131 @@
+```bash
+"${PODS_ROOT}/SwiftLint/swiftlint"
+```
+<img width="1200" alt="build-phaseimg" src="https://user-images.githubusercontent.com/6582396/166877073-ec3594d3-c58b-4b4c-89f8-86529092cab8.png">
````
This image should be adding in the current repository itself

anand00 replied:
updated image URL

Abijeet commented on swift-swiftLint.md:
```diff
@@ -0,0 +1,131 @@
+<img width="1200" alt="build-phaseimg" src="https://user-images.githubusercontent.com/6582396/166877073-ec3594d3-c58b-4b4c-89f8-86529092cab8.png">
+
+
+* Move the ``Run Script`` above the ``Compile Sources`` step by drag and drop, to detect errors quickly before compiling.
```
Use single tilde `

anand00 replied:
updated

anand00 committed:
added Xcode png file

anand00 committed:
Added sugested formate 

1. Removed <br> 
2. Added bullets 1,2,3 in steps
3. Updated image path
4. used single tilde

vishnu-kyatannawar commented on swift-swiftLint.md:
```diff
@@ -0,0 +1,134 @@
+# Introduction
```
Please create a `assets` folder and keep images there.

anand00 replied:
created assets folder and moved images

vishnu-kyatannawar commented on swift-swiftLint.md:
```diff
@@ -0,0 +1,134 @@
+# Introduction
+
+[SwiftLint](https://github.com/realm/SwiftLint/blob/master/README.md) is a tool by Realm to enforce Swift style and conventions. SwiftLint hooks into Apple’s own SourceKit framework to parse your Swift code, which means it supports the full range of Swift syntax and remains up to date as Swift continues to evolve. It is based on guidelines from the Swift style guide.
```
We need to add swift from TODO to Technologies list in README.md.

![image](https://user-images.githubusercontent.com/22377238/168998999-c6661614-0571-4700-93d1-e456ccfd4636.png)

vishnu-kyatannawar replied:
Please ensure to add version number like other technologies.

anand00 replied:
updated

vishnu-kyatannawar commented on swift-swiftLint.md:
```diff
@@ -0,0 +1,134 @@
+# Introduction
```
Please, rename this file name to `swift-swiftlint.md` (all small case). We can standadize a convention for file name later on.

anand00 replied:
file name updated

vishnu-kyatannawar commented on swift-swiftLint.md:
````diff
@@ -0,0 +1,134 @@
+brew install swiftlint
+```
+
+### Using [CocoaPods](https://cocoapods.org):
````
We should standadize to use 1 thing. Either `homebrew` or `cocapods`.

If they are like `npm` and `yarn`. We decided to go with just `npm`, then you need to pick the good and reliable one and only use that.

anand00 replied:
- Removed Carthage option.
- Recommended CocoaPods

anand00 committed:
Created assets directory

OSMAnand committed:
Move image files to /assets

anand00 committed:
Moved swift from TODO to Technologies

anand00 committed:
Rename swift-swiftLint to swift-swift lint.md

anand00 committed:
update swift-swiftlint file name

anand00 committed:
Remove Carthage installation 

1. Recommend installation using cocoapods as per standard.
2. Update image path

anand00 committed:
Update README.md
